Source/DafnyLanguageServer/Workspace/TextDocumentLoader.cs

```diff
@@ -141,29 +148,61 @@ bool loadCanceled
       );
     }
 
-    public Task<DafnyDocument> VerifyAsync(DafnyDocument document, CancellationToken cancellationToken) {
+    public IObservable<DafnyDocument> Verify(DafnyDocument document, CancellationToken cancellationToken) {
+      notificationPublisher.SendStatusNotification(document.TextDocumentItem, CompilationStatus.VerificationStarted);
+      var progressReporter = new VerificationProgressReporter(document.TextDocumentItem, notificationPublisher);
+      var programErrorReporter = new DiagnosticErrorReporter(document.Uri);
+      document.Program.Reporter = programErrorReporter;
+      var implementationTasks = verifier.Verify(document.Program, progressReporter, cancellationToken);
+      foreach (var implementationTask in implementationTasks) {
+        implementationTask.Run();
+      }
+
+      var _ = NotifyStatusAsync(document.TextDocumentItem, implementationTasks);
```

Comment:
Interesting way to start a "thread", I never called an async method without await before.

Reply:
Jep, it's a "fire and forget" IO task. I think you might find something similar inside a call to write to a log.